cmd/spire-server/cli/entry/util_test.go

```diff
@@ -45,3 +49,93 @@ func selectorToFlag(t *testing.T, selectors []*common.Selector) StringsFlag {
 
 	return resp
 }
+
+func TestRegisterParseFile(t *testing.T) {
+	p := path.Join(util.ProjectRoot(), "test/fixture/registration/good.json")
+	entries, err := ParseFile(p)
+	require.NoError(t, err)
+
+	entry1 := &common.RegistrationEntry{
+		Selectors: []*common.Selector{
+			{
+				Type:  "unix",
+				Value: "uid:1111",
+			},
+		},
+		SpiffeId: "spiffe://example.org/Blog",
+		ParentId: "spiffe://example.org/spire/agent/join_token/TokenBlog",
+		Ttl:      200,
+		Admin:    true,
+	}
+	entry2 := &common.RegistrationEntry{
+		Selectors: []*common.Selector{
+			{
+				Type:  "unix",
+				Value: "uid:1111",
+			},
+		},
+		SpiffeId: "spiffe://example.org/Database",
+		ParentId: "spiffe://example.org/spire/agent/join_token/TokenDatabase",
+		Ttl:      200,
+	}
+
+	expectedEntries := []*common.RegistrationEntry{
+		entry1,
+		entry2,
+	}
+	assert.Equal(t, expectedEntries, entries)
+}
+
+func TestParseFileFromStdin(t *testing.T) {
+	p := path.Join(util.ProjectRoot(), "test/fixture/registration/good.json")
+	data, err := ioutil.ReadFile(p)
+	assert.NoError(t, err)
+
+	fakeStdin, err := ioutil.TempFile("", "spire-test")
+	assert.NoError(t, err)
+	// Replace os.Stdin temporary
+	orgStdin := os.Stdin
+	os.Stdin = fakeStdin
```

Comment:
I'd prefer not to override the `os.Stdin` package global in tests.

One option is to use the dependency injection and test the `parseFile(io.Reader, filename)` function and leave `ParseFile(filename)` without directly testing it.

Another option is to set `var stdin = os.Stdin` and then replace the package local `stdin` var, which is much less invasive than mutating `os.Stdin` entirely.

Reply:
587ac9c
This commit changes the target and tests the `parseFile(io.Reader, filename)` function.